### ISSUE TITLE
YAML with arrays are not flattened

### ISSUE THREAD

**Issue by kronn** [henrik/vim-yaml-flattener#6]
The plugin works great for the "hash-only" yaml-files of rails' translations. (thank you for that)

I noticed that it does not work with files that contain arrays. It flattens until it encounters an array but does not flatten the hashes in the array/s.

Since all ops work these days is basically yaml-work (https://twitter.com/rjw1/status/961225355490099201), I would like to use the plugin also for yaml containing arrays. 

One could flatten an array-yaml as follows:

```yaml
---
list:
  - name: foo
  - name: bar
  - name: baz
```

```yaml
---
list[].name: foo
list[].name: bar
list[].name: baz
```

What do you think?

**Comment by henrik** [comment]
Hi! Sorry for not replying sooner; I've been lazy :)

I like that idea! I'm not in a place where I want to spend much time on this plugin, but if anyone wants to make a pull request for this, I'd be happy to merge that!

**Comment by kronn** [comment]
How dare you have a life? And proper priorities? Outrageous! :wink: :joy: 

On a more serious note: Since you're basically open to an extension, I'll put it on my list. The next time I have to work with such a yaml-file, I'll give it a shot. For this, I am VERY happy that the real functionality is implement in Ruby, not VimScript :-) Could happen next week or in the next two month, so don't hold your breath. Just the sorting alone (as accidental as I may be) is a main feature for me.

**Comment by henrik** [comment]
I know! :D 

And sure thing. I'll leave this issue for now, but may close it in several months if there's no activity, to keep things tidy. Contributions will still be welcome even if it's closed, of course.

By the way, I haven't used this plugin myself for years now, instead using ([my fork](https://github.com/henrik/vim-yaml-helper) of) [vim-yaml-helper](https://github.com/lmeijvogel/vim-yaml-helper) which lets me do stuff like `:YamlGoToKey foo.bar.baz` or `YamlFullPath` to see (and copy) the full "foo.bar.baz" string when at `foo:\n  bar:\n    baz: "boink"` in a YAML file.

Last time I checked, the original plugin had inconvenient command names that were harder to abbreviate in Vim, thus my fork.

**Comment by kronn** [comment]
Fair enough. If I haven't gotten around to do it in a few months, any predictions are difficult.

I will take a look at the other plugin. I think I had the original one in use shortly, but cleaned it with other unused ones a while later. I will give your fork a spin and see how it works for me.
